app/controllers/collections_controller.rb

```diff
@@ -1,12 +1,17 @@
 class CollectionsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[index show]
+
   def index
     @user = User.find_by!(username: params[:username])
-    @collections = @user.collections.order(created_at: :desc)
+    @collections = @user.collections.includes(:articles).order(created_at: :desc)
+    @articles = Article.published.where(collection_id: @collections.pluck(:id))
```

Comment:
**suggestion (non-blocking):** Based on @rhymes recent PR, I think you may be able to use `@collections.ids` here.

Reply:
Yeah, theoretically rubocop should have flagged this

Reply:
Shouldn't we use `Collection.table_key` here? The collections are going to be updated when the article is updated:

https://github.com/forem/forem/blob/6ac637b9ef4c74e09201065abe662e2d1e9e8952/app/models/article.rb#L90

https://github.com/forem/forem/blob/6ac637b9ef4c74e09201065abe662e2d1e9e8952/app/models/article.rb#L672-L674